src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -147,8 +273,41 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submiss
     );
   }, [columnFilters, globalFilter, dispatch]);
 
+  // tells page to ask for new entries from main page when its state changes.
+  useEffect(() => {
+    updateEntries(limitGroup, pageParams, backendFilterParams);
+    // updateEntries seems to change between renders as a reference?
```

Comment:
Yes, and as a result, it should be causing an infinite loop if you add it to the dependency array below.

And it should (cause the infinite loop) – this should point to something wrong in your code instead of disabling the warning.

In this case, your `updateEntries` comes from `updateGradingOverviewsCallback` in the `Grading` component, which is not memoized using `useCallback`. Therefore, when `updateEntries` is called and a new array for the grading overviews is made, it will trigger a rerender of `Grading` (as it should). But because `updateGradingOverviewsCallback` is not memoized, it also recreates this function, causing the reference change, causing the prop change, causing `useEffect` to fire again, repeating the whole cycle, resulting in infinite re-renders.

Reply:
Note that I didn't actually run this PR and just read through the code changes briefly, so please take this with a grain of salt and let me know if memoizing `updateGradingOverviewsCallback` doesn't solve the issue.

Reply:
Implemented! Makes sense that the function reference changes for each re-render.